### DISCUSSION TITLE
A few ideas for a more extensive work

### DISCUSSION THREAD

**Discussion by ghost** [alagrede/znote-app#47] [Ideas]
Hellow,

after playing around with zNote during my daily work I noticed a few missing features compared to other tools, like Obsidian or even Typora. And since I really like the environment and look&feel of zNote (and the JavaScript playground, of course), I would love to see this features in here as well.

1. Opening multiple Markdown files at once isn't possible right now, neither by using drag 'n' drop (which also doesn't work with a single file) nor using the "open" file dialog. Would be pretty cool supporting this or point 2.
2. It would be pretty helpful to link a notebook with a local file path. In our company it's common to write a small markdown documentation on larger projects, which is written within the same git repository. Thus, it would be cool 
   1. Linking the notebook with the filepath so I can view and edit all files within this directory
   2. See the relative file path to the file next to the filename (many files are called index.md due to our documentation engine)
   3. Including a meta header, our files usually starts with a few `key: value` pairs (one per line) and ends with three dashes `---` until the final document starts (but that's a more specific feature, I guess).
3. When I select a (new) notebook and **open** a file it is not assigned to this notebook directly, instead it just appears in "All notes". Would be really comfortable to assign this file directly like newly created ones as well.

A plugin API would be really nice as well, of course, but I can imagine that this isn't as easy to implement though.

Anyways, It already looks like a really good alternative, which is a bit less "closed" as Obsidian, but more project aimed as Typora (at least it feels like this). And the JavaScript playground is really helpful in my current project.

Thanks for your work and your product.

~Riley

**Comment by alagrede** [comment]
Many thanks for sharing your feedback @RileysAgony 😊 It is very helpful to help me to move forward.
I will rearrange these points into my backlog for next releases.

Anthony

**Comment by alagrede** [comment]
Hi @RileysAgony,
Just a note to let you know that I have released a new version with some bug fixes and improvements you mentioned above, such as opening multiple files or automatically setting the selected tag and folder when opening the file.
Concerning the meta headers, it's already possible to add yaml (Github like) meta headers (see documentation below):
https://doc.notebookjs.app/#/guide?id=yaml-metadata

See you soon for the next release. 😊
Anthony
